docs/source/pages/using.rst

```diff
@@ -90,6 +90,50 @@ To analyze a repository on a self-hosted GitLab instance, you need to do the fol
 
 - Obtain a GitLab access token having at least the ``read_repository`` permission and store it into the ``MCN_SELF_HOSTED_GITLAB_TOKEN`` environment variable. For more detailed instructions, see `GitLab documentation <https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#create-a-personal-access-token>`_.
 
+''''''''''''''''''''''''''''''''''''''''''''''''''''
+Providing a PURL string instead of a repository path
+''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+Instead of providing the repository path for the analysis software component, a `PURL <https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst>`_. string
+for the target git repository could be used.
+
+To begin with, we follow the same steps for the configuration if needed (especially for self-hosted GitLab instances). The PURL string for a git repository could be created using the following format:
+
+.. code-block::
+
+  pkg:<git_service_domain>/<organization>/<name>
+
+The list bellow shows examples for the corresponding PURL string for different git repositories:
+
+.. list-table:: Example of PURL strings for git repositories.
+   :widths: 50 50
+   :header-rows: 1
+
+   * - Repository path
+     - PURL string
+   * - ``https://github.com/micronaut-projects/micronaut-core``
+     - Both ``pkg:github/micronaut-projects/micronaut-core`` and ``pkg:github.com/micronaut-projects/micronaut-core`` are applicable as ``github`` is a pre-defined type as mentioned `here <https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst>`_.
+   * - ``https://bitbucket.org/snakeyaml/snakeyaml``
+     - Both ``pkg:github/micronaut-projects/micronaut-core`` and ``pkg:github.com/micronaut-projects/micronaut-core`` are applicable as ``bitbucket`` is a pre-defined type as mentioned `here <https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst>`_.
+   * - ``https://internal.gitlab.com/foo/bar``
+     - ``pkg:internal.gitlab.com/foo/bar``
+   * - ``https://gitlab.com/gitlab-org/gitlab``
+     - ``pkg:gitlab.com/gitlab-org/gitlab``
+
+After the PURL string has been obtained, the analysis could be run with:
+
+.. code-block:: shell
+
+  ./run_macaron.sh analyze -purl <purl_string>
+
+You can also provide the PURL string together with the repository path. In this case, the PURL string will be used as the unique identifier to the analysis target:
+
+.. code-block:: shell
+
+  ./run_macaron.sh analyze -purl <purl_string> -rp <repo_path> -b <branch> -d <digest>
+
+.. note:: When provide the PURL and the repository path, both branch name and commit digest must be provided as well.
```

Comment:
```suggestion
.. note:: When providing the PURL and the repository path, both the branch name and commit digest must be provided as well.
```

Reply:
Fixed in https://github.com/oracle/macaron/pull/401/commits/7a84f6f91fe54da57436ef5ac9e065a913932769